### DISCUSSION TITLE
More Example (easy for home farming startup), and List of supported Hardware.

### DISCUSSION THREAD

**Discussion by dzuddiyn** [timmbogner/Farm-Data-Relay-System#223] [General]
Hi Tim,
I go here because of Andreas.

I love your project. but it's hard to start. I'm not a programmer. just got some basic in electronics.
I hope to have;

1. more Example as it would be easy for home/small farming startup like me. 
and
2. List of supported Hardware is important, since I scare to buy wrong boards. - sensor node/gateway boards & supported LoRa modules.

I'm interested in using LoRa, not much on ESP-Now as I reviewed on YT, it only can reach about 100 meters node-to-repeater/gateway. Unfortunately, there are only few all-in-one ready made good LoRa sensor nodes available, especially with good deep sleep. 

After several days of searching, I found Heltec sensor node. BUT sadly, unfortunately, I bought CubeCell sensor nodes (not ESP based) (buy because it sleeps at 3.5 uA while ESP base mostly 70 uA).

BUT then. I just realizes that it is not supported by FRDS, really sad. After go through reading a long discussion here, I found your statement that the HelTec LoRa 32(V3) (https://heltec.org/project/wifi-lora-32-v3/) is supported, double sadness as I should buy Heltec ESP32 base modules, also good complete sensor node too. but in fact, I still doubt it as there are no official hardware list in wiki & no example project.

my request are;

1. may you upload your code of sensor nodes (+) the hardware you use. hope as many as you can. so we can duplicate and use it right away or just minimal edit. and hope some documents of the hardware setup picture, as you did in soil sensor example, but without the casing. I thinks no need long words, just picture of the hardware and its name (we can google for more info), so we could have idea. and please state which line in code to define pin / communication (I2C/SPI/UART) to the sensors / sensor modules.
2. please list down the supported hardware name (no need link as we can just google it).
3. and LoRa module that supported by FRDS such as Ai Thinker/eByte/etc.

If there are several sensor node examples in this github documentation Or example projects in Hackster / Instructable, this project would be a boom, as it is easy to kickstart, just buy supported hardware, pin-in the sensors and burn-in the firmware code, boom, everyone got IoT project to spare food security on ourselves. Your Karma of kindness would exponential.

THANKS in advance Mr Tim, an intelligent farmer.

**Comment by timmbogner** [comment]
Hello @dzuddiyn ,
Thanks a lot for checking out FDRS! 
There are a lot of different sensor examples in the `Farm-Data-Relay-System/examples` directory of the repo. You will find generic ESP-NOW and LoRa sensor sketches, which send sample "dummy" sensor readings every ten seconds. These are where you should begin your testing. The `LoRa_Sensor` sketch will be received by the `1_UART_Gateway` sketch if `USE_LORA` is enabled in the gateway configuration. You will also find several examples for commonly used environmental sensors in the `Farm-Data-Relay-System/examples/Sensor_Examples`. Let me know which other examples you'd be looking for and I'll see what I can do.

Any ESP8266 or ESP32 device should be compatible. If the device has an ESP microcontroller attached to a Semtech SX127* or SX126* LoRa radio, it will work. Anything on LilyGo or Heltec's website that says it has an ESP32 and LoRa will work. Cubecell is reported to work, but there are some extra steps. Raspberry Pi Pico with a LoRa hat made by Waveshare has been reported to work. 

It's a little too broad to maintain a list. You're welcome to ask me here on GitHub before buying a module and I'll double-check the schematics for you. Sorry to hear of your CubeCell woes. Amazon's fooled lots of people into thinking they were getting an ESP32 with those.

**Reply by dzuddiyn** [reply]
thanks for your kind guide..

may I know 1 thing.. from your broad experience..

what are the **recommended LoRa** module **best for FRDS**?
perhaps 1 recommendation for **all-in-one** module **& 1** recommendation for **DIY version** (the one which could **sleep tight in uA**)

So I just buy, and modify a little from your sensor node examples.

also if not bothering you.
what are the **recommended** **all-in-one** **ESP-Now** module best for FRDS? **+ 1** recommendation for **DIY** version.

thanks in advance!!

**Reply by timmbogner** [reply]
To develop the LoRa functionality, I used LilyGo's LoRa32 v1.0 boards. These are now sold out, so I would recommend [the 1.3 ](https://www.lilygo.cc/products/lora-v1-3) boards.

There are so many different modules out there that use ESP32, so I'd honestly search out the one that fulfills your needs. I buy lots of generic ESP8266 Wemos D1 Mini to experiment with, and they are cheap and replaceable. Great for nodes.
I would recommend ESP32 for gateways. [LilyGo's T-Display ](https://www.lilygo.cc/products/lilygo%C2%AE-ttgo-t-display-1-14-inch-lcd-esp32-control-board)has been my go-to board for a generic ESP32 device.

If you are looking for a solution with two ESP32s in one, look at the Thingpulse ESPGateway I discuss in the main readme.

**Reply by dzuddiyn** [reply]
thanks for the clarification..

**Comment by aviateur17** [comment]
I'm thinking as a future possibility maybe develop .bin files for specific boards so that people would just flash the .bin file to the board without having to change any config files?  Maybe have that ability via WiFi?   Initial flash would be via USB Serial with .bin file and then future flashes (ideally) would be via ESP32 OTA.  This could get them started and then as they gain confidence they can start editing the config files for different board types.   Maybe create a bunch of different examples with specific config files for different specific board types.  Just shooting out some ideas.

**Reply by dzuddiyn** [reply]
That sounds like an even cool idea! 

I'm not entirely grasp the idea, but it seems like it would be a simple software or webpage similar to Meshtastic that can be easily set up.
or
we could choose different boards and settings from a dropdown menu in Arduino IDE and have the right configuration uploaded to the hardware automatically based on the wiring instructions provided. 

It would be awesome to have someone help out with this!

**Reply by timmbogner** [reply]
I like where this is going, but I'm not sure if using the .bin file will be easier than Arduino IDE for most beginners. However, if we could allow users to flash their devices from a website, that would be my dream. I know that WLED has this ability, maybe Meshtastic too. If anyone knows the secret to this magic, please chime in! Sometime soon I want to discuss dependencies and whether we can include RadioLib, ArduinoJson, and Pubsubclient with the repo, which would probably be the first step to that.

EDIT: Or what about console scripts with Arduino CLI? 
A UI-based config file editor application sounds like a relatively simple concept, probably possible through Javascript (?). Then we'd just need to automatically compile and flash, not super-complicated. We could also have pre-compiled example sketches available with a click.
